build/makelib/golang.mk

```diff
@@ -170,7 +170,14 @@ go.mod.update:
 
 .PHONY: go.mod.check
 go.mod.check:
-	@echo === ensuring modules are tidied
+	@echo === syncing root modules with APIs modules
+	@cp -a go.mod pkg/apis/go.mod
+	@cp -a go.sum pkg/apis/go.sum
+	@sed -i 's|^module github.com/rook/rook|module github.com/rook/rook/pkg/apis|' pkg/apis/go.mod
```

Comment:
`sed -i` have an issue with Mac that's why we are seeing the build issue.

I did this and passed that error stage 
```git diff
diff --git a/build/makelib/golang.mk b/build/makelib/golang.mk
index a18c64ffc..3cb7de8c1 100644
--- a/build/makelib/golang.mk
+++ b/build/makelib/golang.mk
@@ -173,8 +173,8 @@ go.mod.check:
        @echo === syncing root modules with APIs modules
        @cp -a go.mod pkg/apis/go.mod
        @cp -a go.sum pkg/apis/go.sum
-       @sed -i 's|^module github.com/rook/rook|module github.com/rook/rook/pkg/apis|' pkg/apis/go.mod
-       @sed -i '\:^replace github.com/rook/rook/pkg/apis => ./pkg/apis:d' pkg/apis/go.mod
+       @sed -i '' 's|^module github.com/rook/rook|module github.com/rook/rook/pkg/apis|' pkg/apis/go.mod
+       @sed -i '' '\:^replace github.com/rook/rook/pkg/apis => ./pkg/apis:d' pkg/apis/go.mod
        @echo === ensuring APIs modules are tidied
        @(cd pkg/apis/; $(GOHOST) mod tidy -compat=1.17)
        @echo === ensuring root modules are tidied
```

Reply:
This way it wound work on Linux

Reply:
fixed for both Linux and Mac